README.md

```diff
@@ -1,634 +1,1828 @@
+may take ANOTHER few days to settle. Do you see where this is going? By the time all of the settling
+has finished, the opportunity may have long passed.
+
+While the above example doesn't include Cardano-Loans, the point is that composition allows for
+creating truly complex economic transactions from very simple building blocks. Just like with
+Wolfram's cellular automaton, it is very hard to predict what the ultimate economy will look like.
+Because of this, **all core DeFi DApps should priortize composability**. People may come up with
+useful compositions that the DApp developers never considered. The more compositions that are
+possible, the larger, and more complex the DeFi economy can become. And as the risk-management
+example shows, composability can make the complex DeFi economy more resilient.
+
+### No Trustless Credit History
+
+You simply cannot have a healthy credit-debt market without credit history. However, if a market
+does have credit history but that history is not trustless, it just creates another attack vector
+governments and hackers can use to target the growing DeFi economy. 
+
+Some DApps try to rely on off-chain credit history databases, but this is not a good approach.
+*Reading* credit histories must be just as trustless as *writing* credit histories. Even if the
+blockchain records the credit history trustlessly, if lenders can't query the credit history, it is
+effectively the same as there being no credit history. A centralized database, or a database that is
+hard to read, is easy to commandeer/censor. 
+
+In practical terms, if only beefy computers are able to read the credit histories from the
+blockchain, those with these computers are in a priviledged position between borrowers and lenders.
+They can deny sharing the credit histories for borrowers they don't like. This would effectively
+nullify a borrower's entire credit history since lenders will most likely treat an unknown history
+the same as no history. Even if the people with these beefy computers are honest, they are a very
+centralized group that may be easy for governments target. If a government is able to commandeer
+these beefy computers, governments will still be able to exert undue influence. Blockchain would
+fail to be a check on government abuse. This argument applies to priorietary credit history
+databases as well.
+
+Credit histories *must* be on-chain and *must* be easy to lookup, even with an average computer.
+Otherwise, the credit history database itself would be an existential risk for a DeFi credit-debt
+market.
+
+> NOTE: Data availability is still a work in progress, but so far, there are no alternatives to
+> storing the critical information on-chain. There may never be an alternative; the whole point of
+> the layer 1 chain is to create a global database of critical information. Custody of money is
+> *not* the only critical information an economy relies on.
+
+### Centralized
```

Comment:
Might also add to this section this link w/ short introduction to changes already in progress: https://cointelegraph.com/news/upcoming-defi-rules-europe-could-ban-non-decentralized-protocols

Reply:
I added a paragraph that conncts to the link. I also added a comment that a "truly decentralized DApp should be *legally unregulatable*". I like this phrase and believe it is true even though I am not 100% sure it is true. Perhaps there is still some applicable regulation even though I can't picture what that would be...